src/test/java/core/basesyntax/service/fileoperation/impl/CsvFileWriterImplTest.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.service.fileoperation.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.FruitShopStorage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.fileoperation.CreateReport;
+import core.basesyntax.service.fileoperation.CsvFileWriter;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.AfterClass;
+import org.junit.Test;
+
+public class CsvFileWriterImplTest {
+    private static final String OUTPUT_SOURCE = "src/test/resources/test-output.csv";
+    private static CreateReport report;
+    private static CsvFileWriter writer;
+
+    @AfterClass
+    public static void clear_storage() {
+        FruitShopStorage.storageFruits.clear();
+    }
+
+    @Test
+    public void write_report_Ok() {
+        StorageDao dao = new StorageDaoImpl();
+        writer = new CsvFileWriterImpl();
+        report = new CreateReportImpl(dao);
+        dao.addFruit(new Fruit("apple", 40));
+        dao.addFruit(new Fruit("peach", 50));
+        dao.addFruit(new Fruit("apricot", 10));
+        writer.writeFile(OUTPUT_SOURCE, report.getReport());
+        List<String> actual = new CsvFileReaderImpl().inputFile(OUTPUT_SOURCE);
+        List<String> expected = dao.getAll()
+                .stream()
+                .map(s -> s.getName() + "," + s.getAmountFruit())
+                .collect(Collectors.toList());
+        String title = "fruit,quantity";
+        expected.add(0, title);
+        assertEquals(expected, actual);
```

Comment:
To check if the recording is correct, create a reader in the same way you created a writer in the test where you tested the reader. And use it to check if the data is correctly written to the file